src/bug-filing/azure-boards/azure-boards-bug-filing-service.tsx

```diff
@@ -1,52 +1,52 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 import { isEmpty } from 'lodash';
-import * as React from 'react';
-
+import { IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
 import { EnvironmentInfo } from '../../common/environment-info-provider';
-import { NamedSFC } from '../../common/react/named-sfc';
 import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
 import { BugServicePropertiesMap } from '../../common/types/store-data/user-configuration-store';
 import { BugFilingService } from '../types/bug-filing-service';
-import { SettingsFormProps } from '../types/settings-form-props';
+import { AzureBoardsSettingsForm } from './azure-boards-settings-form';
+
+const AzureDevOpsBugFilingServiceKey = 'azureBoards';
 
-const AzureDevOpsBugFilingServiceKey = 'azureDevOps';
+export type AzureBoardsIssueDetailField = 'reproSteps' | 'description';
 
-export type AzureDevOpsBugFilingSettings = {
+export interface AzureBoardsIssueDetailLocationDropdownOption extends IDropdownOption {
+    key: AzureBoardsIssueDetailField;
+}
+
+export type AzureBoardsBugFilingSettings = {
     projectURL: string;
-    issueDetailsLocationField: string;
+    issueDetailsField: AzureBoardsIssueDetailField;
 };
 
-function buildStoreData(projectURL: string, issueDetailsLocationField: string): AzureDevOpsBugFilingSettings {
+function buildStoreData(projectURL: string, issueDetailsField: AzureBoardsIssueDetailField): AzureBoardsBugFilingSettings {
     return {
         projectURL,
-        issueDetailsLocationField,
+        issueDetailsField,
     };
 }
 
-function getSettingsFromStoreData(bugServicePropertiesMap: BugServicePropertiesMap): AzureDevOpsBugFilingSettings {
-    return bugServicePropertiesMap[AzureDevOpsBugFilingServiceKey] as AzureDevOpsBugFilingSettings;
+function getSettingsFromStoreData(bugServicePropertiesMap: BugServicePropertiesMap): AzureBoardsBugFilingSettings {
```

Comment:
Comparing to getSettingsFromStoreData in github-bug-filing-service.tsx, two thoughts:
1. The parameter type in that function is not correct (it's `GitHubBugFilingSettings`, should be `BugServicePropertiesMap`)
2. Except for return type, it's the same function - does every service need to implement this?

Reply:
For 2: yes, every service need to implement the same function. We talked about having a builder to put the same function on every service object, but feel like it is a bit over engineering for now.

Reply:
As a middle ground solution: extracted the "duplicated" code to a creational function. As Haonan mentioned, we discussed several options to facilitate the creation of new issue filing services but decide against it at this point as there is a lot of refactoring going on which make it a tad hard to visualize the end result. And as with other pieces of this feature, we'll introduce some better way to handle this if we needed as part of our clean up work at the end of the feature.